yaml/shared/src/main/scala/org/virtuslab/yaml/NodeVisitor.scala

```diff
@@ -0,0 +1,100 @@
+package org.virtuslab.yaml
+
+import org.virtuslab.yaml.Node
+import org.virtuslab.yaml.Node.*
+import org.virtuslab.yaml.Range
+import org.virtuslab.yaml.TraverseError
+import org.virtuslab.yaml.YamlError
+import org.virtuslab.yaml.syntax.NodeSelector
+import org.virtuslab.yaml.syntax.NodeSelector.*
+
+case class NodeVisitor(node: Node, selectors: List[NodeSelector]):
```

Comment:
```suggestion
class NodeVisitor private (node: Node, selectors: List[NodeSelector]):
```
+ `def apply(node: Node, selectors: List[NodeSelector]): NodeVisitor = new NodeVisitor(node, selectors)` in companion object
There is no reason for `NodeVisitor` to be a case class. Moreover, it exposes copy method.